bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManagerImpl.java

```diff
@@ -1111,6 +1128,61 @@ private void setThingStatus(Thing thing, ThingStatusInfo thingStatusInfo) {
         }
     }
 
+    @Override
+    public void enableThing(ThingUID thingUID, boolean enabled) {
```

Comment:
Check if you can do it as closely as possible to [the rule engine](https://github.com/kaikreuzer/smarthome/blob/circular_dependencies/bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngineImpl.java#L873-L889) - it currently looks wrong to me that you get hold of the `thingHandler` as this should actually never be set for a disabled thing.